trestle/core/commands/init.py

```diff
@@ -0,0 +1,66 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Trestle Init Command."""
+
+import os
+from shutil import copyfile
+
+from ilcli import Command
+
+from pkg_resources import resource_filename
+
+import trestle.core.const as const
+
+
+class InitCmd(Command):
+    """Initialize a Trestle working directory."""
+
+    name = 'init'
+
+    def _run(self, args):
+        """Create a trestle project in the current directory."""
+        dir_path = os.getcwd()
+
+        try:
+            # Create directories
+            self._create_directories()
+
+            # Create config file
+            self._copy_config_file()
+
+            self.out(f'Initialized trestle project successfully in {dir_path}')
+
+        except BaseException as err:
+            self.err(f'Initialization failed: {err}')
+            return 1
+
+    def _create_directories(self):
+        """Create the directory tree if it does not exist."""
+        # Prepare directory list to be created
+        directory_list = [const.TRESTLE_CONFIG_DIR]
+        for model_dir in const.TRESTLE_MODEL_DIRS:
+            directory_list.append(model_dir)
+            directory_list.append(os.path.join('dist', model_dir))
+
+        # Create directories
+        for directory in directory_list:
+            os.makedirs(name=directory, mode=0o7777, exist_ok=True)
```

Comment:
Should we be specifiying the mode or inheriting it based on user config?

Reply:
Removed mode= parameter and relying of current umask.